### PR TITLE
Fix downloading pdf produces html as extension

### LINKED ISSUE #4913: Downloads from the arXiv are marked as HTML files instead of PDF
JabRef version: latest  on  Windows

- [x] I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. Import an entry from the arXiv, say `1207.0408v1` (using Library > New Entry > ArXiv)
2. Go to File tab, right-click link, and select Download
3. The file is downloaded successfully, but has file extension `html` instead of `pdf`. Renaming the file results in a readable pdf through.

<details>
  <summary>Log File</summary>
  
  ```
  Paste an excerpt of your log file here
  ```
</details>

### PR DESCRIPTION
**Pull request by Siedlerchr**
If we already have a filetype we use that instead of relying on the autodetection

Fixes #4913

----

- [ ] Change in CHANGELOG.md described
- [x] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [x] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/gui/filelist/LinkedFilesEditDialogViewModel.java [hunk 1]
```diff
@@ -91,9 +91,7 @@ public void openBrowseDialog() {
 
     public void setValues(LinkedFile linkedFile) {
         description.set(linkedFile.getDescription());
-
-        Path linkPath = Paths.get(linkedFile.getLink());
-        link.set(relativize(linkPath));
+        link.setValue(linkedFile.getLink()); //Might be an URL
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/filelist/LinkedFilesEditDialogViewModel.java]: Now the `relativize` command is missing, right?
Maybe have something like `if (linkedFile.isOnline()) link.setValue(...) else link.setValue(relativize(Paths.get(...)))`?

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/filelist/LinkedFilesEditDialogViewModel.java]: The relativize command was creating an error, if the url is a string.
I also tested the rename functionality and it's fine, no need for the relativizere here I think

Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/filelist/LinkedFilesEditDialogViewModel.java]: The relativize is only for the display, and I would prefer to keep it.
I was proposing the if (linkedFile.isOnline()) check to see if the string is an url.

**Comment by Siedlerchr** [comment]
I tried adding a test, but it's a bit complicated, as the logic is an lambda in the prepareDownloadTask method. And solely testing the new method which just passes the type as parameter is trivial.

**Comment by tobiasdiez** [comment]
Mhh, you should be able to test the `download` method directly:
```java
LinkedFile onlineFile = new LinkedFile("http://arxiv something");
LinkedFileViewModel viewModel = new LinkedFileViewModel(onlineFile, ....);
viewModel.download();
assertEquals("expectedPath", onlineFile.getLink());
```
needs a bit more setup, but should work in principle.

**Comment by Siedlerchr** [comment]
Got it to work with heavy mocking and refactoring. The shitty externalFileTypes Singleton is really bad for testing.

**Pull request merged**